### PR TITLE
(DOCSP-10330): Task Tracker (Web)

### PR DESCRIPTION
**Pull request by nlarew**
**Tutorial Docs PR: https://github.com/mongodb/docs-realm/pull/210**

In general, the code in `/src/realm` is what the user interacts with. The files in `/src/components` and `/src/hooks` are meant to be code complete so that users can dive in but don't need to know React.

## TODO

- Remove my test App ID and replace it with ``"<Your App ID>"`
- Remove login helper code (e.g. `joe.jasper@example.com` is the default email input value)
- Replace code that users are supposed to write in `/src/realm` with `// TODO: ...` comments
- Change `realm-web` to point to npm rather than `static.realm.io`
- Update links from `realm-dev.mongodb.com` to `realm.mongodb.com`

### CODE REVIEW

File: web/src/graphql-operations.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      }
+export function useGetAllTasksLazyQuery(baseOptions?: ApolloReactHooks.LazyQueryHookOptions<Types.GetAllTasksQuery, Types.GetAllTasksQueryVariables>) {
+          return ApolloReactHooks.useLazyQuery<Types.GetAllTasksQuery, Types.GetAllTasksQueryVariables>(GetAllTasksDocument, baseOptions);
+        }
+export type GetAllTasksQueryHookResult = ReturnType<typeof useGetAllTasksQuery>;
+export type GetAllTasksLazyQueryHookResult = ReturnType<typeof useGetAllTasksLazyQuery>;
+export type GetAllTasksQueryResult = ApolloReactCommon.QueryResult<Types.GetAllTasksQuery, Types.GetAllTasksQueryVariables>;
+export const AddTaskDocument = gql`
+    mutation AddTask($task: TaskInsertInput!) {
+  task: insertOneTask(data: $task) {
```
Comment (cbush) [inline web/src/graphql-operations.ts:50]: Indentation looks a bit off here.

File: web/src/hooks/useDraftTask.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+export default function useDraftTask(
+  taskActions: TaskActions
+): [DraftTask | null, DraftTaskActions] {
+  const [draft, setDraft] = React.useState<DraftTask | null>(null);
+  const draftAssignee = useDraftAssignee(draft);
+  
+  const actions: DraftTaskActions = {
+    createDraft: (draft: DraftTask) => {
+      if(draft) {
```
Comment (cbush) [inline web/src/hooks/useDraftTask.tsx:27]: Formatting
```suggestion
      if (draft) {
```

File: web/src/components/TaskView.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  ],
+  [
+    TaskStatus.Complete,
+    { background: uiColors.green.base, text: uiColors.gray.light3 },
+  ],
+]);
+
+// const KeyID = styled.div`
+//   font-size: 10px;
+// `;
```
[the remark this answers is not in the document]
Comment (nlarew) [reply web/src/components/TaskView.tsx:120]: Nope!

File: web/src/graphql-operations.ts (hunk 1 above)
Comment (nlarew) [reply web/src/graphql-operations.ts:50]: Yeah it's wonky but that's just how the generated code comes out. I guess I could add a step that runs prettier on the generated files 🤔

Comment (cbush) [reply web/src/graphql-operations.ts:50]: Oops, nevermind.

File: web/src/realm/RealmApp.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import * as RealmWeb from "realm-web";
+
+const REALM_APP_ID = "myrealmapp-vjmee";
```
Comment (cbush) [inline web/src/realm/RealmApp.tsx:4]:
```suggestion
const REALM_APP_ID = "<your Realm app ID>";
```

**Pull request merged**